Main_Solution/main_solution_test.rb

```diff
@@ -94,24 +95,66 @@ def test_create_version
   end
 
   def test_create_issue_bug
-
+    register_user
+    create_project
     issue_options = create_issue('bug')
 
-    assert_equal(issue_options[:issue_url_slug], issue_options[:created_issue_url_slug])
+    assert_equal(issue_options[:visible_issue_id], issue_options[:created_issue_id])
   end
 
   def test_create_issue_feature
-
+    register_user
+    create_project
     issue_options = create_issue('feature')
 
-    assert_equal(issue_options[:issue_url_slug], issue_options[:created_issue_url_slug])
+    assert_equal(issue_options[:visible_issue_id], issue_options[:created_issue_id])
   end
 
   def test_create_issue_support
-
+    register_user
+    create_project
     issue_options = create_issue('support')
 
-    assert_equal(issue_options[:issue_url_slug], issue_options[:created_issue_url_slug])
+    assert_equal(issue_options[:visible_issue_id], issue_options[:created_issue_id])
+  end
+
+  def test_conditional_create_issue
```

Comment:
this method should work but don't you think it's too complex here?
we can discuss it )